TheaterSchedule.BLL/Services/ScheduleServices.cs

```diff
@@ -0,0 +1,34 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using TheaterSchedule.DAL;
+using TheaterSchedule.DAL.Interfaces;
+
+namespace TheaterSchedule.BLL
+{
+    public class ScheduleServices : IScheduleServices
+    {
+        private ITheaterScheduleUnitOfWork uow;
+
+        public ScheduleServices(ITheaterScheduleUnitOfWork repo)
+        {
+            uow = repo;
+        }
+
+        public IEnumerable<Schedule> GetListPerformancesByDateRange(DateTime startRange, DateTime endRange)
+        {
+            List<Schedule> listPerfomances = uow.Schedule.GetWithInclude(p => p.performance).ToList();
+            List<Schedule> listPerfomancesDateRange = new List<Schedule>();
+
+            foreach (var perfomance in listPerfomances)
+            {
+                if (perfomance.Beginning >= startRange && perfomance.Beginning <= endRange)
```

Comment:
use iqueryable instead fetching all records from database and filtering in C# code
IQueryable will generate SQL that will select only needed records